rust/worker/src/blockstore/arrow/provider.rs

```diff
@@ -180,14 +180,18 @@ impl BlockManager {
                                 self.read_cache.write().insert(*id, block.clone());
                                 Some(block)
                             }
-                            Err(_) => {
-                                // TODO: log error
+                            Err(e) => {
```

Comment:
I think we should leave this TODO in some fashion, this should return a Result<> IMO